Feature cli

pxp928 opened the pull request:
*Please fill in the fields below to submit a pull request.  The more
information that is provided, the better.*

**Fixes issue #**:


**Description of pull request**:
CLI separated to allow for easier upstreaming. Modifications made to keylib and runlib to add functionality for CLI

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] Tests have been added for the bug fix or new feature
- [x] Docs have been added for the bug fix or new feature

Commit:
added cli

Commit:
added cli

pxp928 commented:
Let me know if there is anything that is missing or if we need to add more comments/unit tests in order for this to upstream.

sudo-bmitch reviewed (approved):
Initial review LGTM. I'd like to add a Makefile, GHA, Readme changes, and do some testing in the lab, but this should be good enough to merge now and iterate with more commits later.

pxp928 commented:
There will be one more branch that will include the spire portion. That would just be the overall "boxboat" fork that would include the Makefile. Should that be added to this branch instead? I can add a ReadMe to this pull request also if needed.

pxp928 commented:
> Initial review LGTM. I'd like to add a Makefile, GHA, Readme changes, and do some testing in the lab, but this should be good enough to merge now and iterate with more commits later.

There will be one more branch that will include the spire portion. That would just be the overall "boxboat" fork that would include the Makefile. Should that be added to this branch instead? I can add a ReadMe to this pull request also if needed.

Commit:
added readme for the cli

pxp928 commented:
> Initial review LGTM. I'd like to add a Makefile, GHA, Readme changes, and do some testing in the lab, but this should be good enough to merge now and iterate with more commits later.

Added new readme for the CLI. Removed `--spiffe-workload-api-path`, `-c, --cert string`, `-i, --intermediate-certs strings` and `--lstrip-paths stringArray` from the documentation as not part of this branch.

pxp928 committed:
updated go.sum entries

pxp928 committed:
updated with upstream master

pxp928 committed:
added certifcate back into cli, updated readme and added makefile

pxp928 committed:
moved layout and openssl into cert folder

sudo-bmitch commented on go.mod:
```diff
@@ -5,12 +5,7 @@ go 1.17
-	golang.org/x/sys v0.0.0-20210616094352-59db8d763f22
-)
-
-require (
-	github.com/davecgh/go-spew v1.1.0 // indirect
-	github.com/pmezard/go-difflib v1.0.0 // indirect
-	gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c // indirect
```
I wonder if this is tripping up the builds. 1.17 added some changes around tracking indirect references that shows up as a separate require section. Try reverting this file, bumping up your local go install to 1.17, and then running `go mod tidy -go=1.17`. For me the file looks like (from an earlier commit):

```
module github.com/in-toto/in-toto-golang

go 1.17

require (
        github.com/codahale/rfc6979 v0.0.0-20141003034818-6a90f24967eb
        github.com/shibumi/go-pathspec v1.2.0
        github.com/spf13/cobra v1.2.1
        github.com/stretchr/testify v1.7.0
        golang.org/x/sys v0.0.0-20210909193231-528a39cd75f3
)

require (
        github.com/davecgh/go-spew v1.1.1 // indirect
        github.com/inconshreveable/mousetrap v1.0.0 // indirect
        github.com/pmezard/go-difflib v1.0.0 // indirect
        github.com/spf13/pflag v1.0.5 // indirect
        gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
)
```

sudo-bmitch replied:
Reference to the 1.17 change: https://golang.org/doc/go1.17#go-command

pxp928 replied:
That worked! Thank you for looking into it. Apparently the path to my local go was pointing to an old install of go on version 1.16. Using 1.17 resolved the issue.

pxp928 committed:
go mod tidy using go version 1.17
